### ISSUE TITLE
Rules refire after going out of the ticket and coming back to it

### ISSUE THREAD

**Issue by githubbob42** [githubbob42/mingle2github2#2804]
**[Mingle Card: 3089](https://liquidframeworks.mingle.thoughtworks.com/projects/alpine_mobile/cards/3089)**
```
| | |
|-|-|
|Card Accepted|

  

 


``````
| | |
|-|-|
|Jenkins QA Job|
|[127](http://eaglepeak:8080/job/Deploy_Mobile_Parameterized/127/)|
|QA Site|
|[https://fxqa1.herokuapp.com](https://fxqa1.herokuapp.com/)|

  

 

``````
| | |
|-|-|
|Jenkins QA Job|
|[126](http://eaglepeak:8080/job/Deploy_Mobile_Parameterized/126/)|
|QA Site|
|[https://fxqa1.herokuapp.com](https://fxqa1.herokuapp.com/)|

  

 


``````
| | |
|-|-|
|Jenkins QA Job|
|[125](http://eaglepeak:8080/job/Deploy_Mobile_Parameterized/125/)|
|QA Site|
|[https://fxqa1.herokuapp.com](https://fxqa1.herokuapp.com/)|

  

 

``````
| | |
|-|-|
|Pull Request:|

   [1334](https://github.com/Liquidframeworks/alpine-mobile/pull/1334)|

  

    |
|[Push to Site fxqa1](http://buildslave3.basecamp.liquidframeworks.com:5000/requestQAbuild?qaSite=fxqa1&repoUrl=https%3A%2F%2Fgithub.com%2FCraigCav%2Falpine-mobile&branch=card3089&repoName=alpine-mobile&repoOwner=Liquidframeworks&cards=%5B%7B%22projectName%22%3A%22alpine_mobile%22%2C%22cardNumber%22%3A%223089%22%7D%5D&requestNumber=1334&requestUrl=https%3A%2F%2Fgithub.com%2FLiquidframeworks%2Falpine-mobile%2Fpull%2F1334)|
|[Push to Site fxqa2](http://buildslave3.basecamp.liquidframeworks.com:5000/requestQAbuild?qaSite=fxqa2&repoUrl=https%3A%2F%2Fgithub.com%2FCraigCav%2Falpine-mobile&branch=card3089&repoName=alpine-mobile&repoOwner=Liquidframeworks&cards=%5B%7B%22projectName%22%3A%22alpine_mobile%22%2C%22cardNumber%22%3A%223089%22%7D%5D&requestNumber=1334&requestUrl=https%3A%2F%2Fgithub.com%2FLiquidframeworks%2Falpine-mobile%2Fpull%2F1334)|
|[Push to Site fxqa3](http://buildslave3.basecamp.liquidframeworks.com:5000/requestQAbuild?qaSite=fxqa3&repoUrl=https%3A%2F%2Fgithub.com%2FCraigCav%2Falpine-mobile&branch=card3089&repoName=alpine-mobile&repoOwner=Liquidframeworks&cards=%5B%7B%22projectName%22%3A%22alpine_mobile%22%2C%22cardNumber%22%3A%223089%22%7D%5D&requestNumber=1334&requestUrl=https%3A%2F%2Fgithub.com%2FLiquidframeworks%2Falpine-mobile%2Fpull%2F1334)|
|[Push to Site fxqa4](http://buildslave3.basecamp.liquidframeworks.com:5000/requestQAbuild?qaSite=fxqa4&repoUrl=https%3A%2F%2Fgithub.com%2FCraigCav%2Falpine-mobile&branch=card3089&repoName=alpine-mobile&repoOwner=Liquidframeworks&cards=%5B%7B%22projectName%22%3A%22alpine_mobile%22%2C%22cardNumber%22%3A%223089%22%7D%5D&requestNumber=1334&requestUrl=https%3A%2F%2Fgithub.com%2FLiquidframeworks%2Falpine-mobile%2Fpull%2F1334)|

  

 


``````
| | |
|-|-|
|Pull Request:|
|[1310](https://github.com/Liquidframeworks/alpine-mobile/pull/1310)|

  

    |
|[Push to Site fxqa1](http://buildslave3.basecamp.liquidframeworks.com:5000/requestQAbuild?qaSite=fxqa1&repoUrl=https%3A%2F%2Fgithub.com%2Fmramad2%2Falpine-mobile&branch=rule-engine-fixes&repoName=alpine-mobile&repoOwner=Liquidframeworks&cards=%5B%7B%22projectName%22%3A%22alpine_mobile%22%2C%22cardNumber%22%3A%223089%22%7D%5D&requestNumber=1310&requestUrl=https%3A%2F%2Fgithub.com%2FLiquidframeworks%2Falpine-mobile%2Fpull%2F1310)|
|[Push to Site fxqa2](http://buildslave3.basecamp.liquidframeworks.com:5000/requestQAbuild?qaSite=fxqa2&repoUrl=https%3A%2F%2Fgithub.com%2Fmramad2%2Falpine-mobile&branch=rule-engine-fixes&repoName=alpine-mobile&repoOwner=Liquidframeworks&cards=%5B%7B%22projectName%22%3A%22alpine_mobile%22%2C%22cardNumber%22%3A%223089%22%7D%5D&requestNumber=1310&requestUrl=https%3A%2F%2Fgithub.com%2FLiquidframeworks%2Falpine-mobile%2Fpull%2F1310)|

  

 

```

### QA Notes

Rule engine added 2 items. Deactivated one rule. Items remained on existing ticket. Created a new ticket and verified one one of the items got added.

Dev Notes
---------

Each rule manages it's pb items, and there is no such thing as shared pb items.Thus, the rule engine will add the item twice because each rule manages it's own item. The fact (that was the result of the bug). 

The fix merely reverts it back to it's correct state(even though we decided not to touch it), and in hindsight, this solution is a far simpler conceptually then having multiple rules manage one item. 

The fix also resolves the issue listed on card #3028.

**The issue also affected the disallow rule. Verify that this issue has been resolved using the disallow rule also.**

Steps to reproduce
------------------

```
| | |
|-|-|
|**Version #**|
|FieldFX 4.0.1.2678|
|**OS**|
|Win|
|**Browser**|
|Chrome|
|**Username**|
|ajf@t3.com|
|**Password**|
|computer47|

  

 

```

1.  I created 2 rules which added the same item and both had the item as required
    
2.  I created the ticket and turned on the rules. 
    
    3\. The item was added only once not 2 times
    
    4\. Then I turned off one rule - it removed the item even though the second rule was still on
    
    5\. I went out of the ticket and came back to the ticket item and the item was re-added based on the rule that was still turned on
    
    6\. If I now turn off that rule then it does not add the item any more.
    

Expected result
---------------

*   Item should be added twice (not clear at the moment)
    *   This behaviour has not been specified in the requirements. Don't do anything about this until we have clear requirements. **(Correction: This behavior is already built in to the rule engine, so the fix will result in pb items being added twice. If future requirements come in limiting this, then we can address it in a seperate card)**
*   If the second rule is still on the item should not have been removed
    *   This is a legitimate defect

Actual Result
-------------

The item is removed and only if you come out of the ticket and come back to the ticket item page the item is readded however, if you turn off the second rule also the item is not added anymore.

Related Cards
-------------

Test Plan
---------

**Test Case: **Two or more rules which add the same item

**Comment by githubbob42** [comment]
### Related Story: #2587 Rule Engine ([Mingle](https://liquidframeworks.mingle.thoughtworks.com/projects/alpine_mobile/cards/2864))

**Comment by githubbob42** [comment]
**Pull Request #1334**
